modules/ROOT/pages/db/db-connector-insert-update-delete.adoc

```diff
@@ -3,48 +3,157 @@ ifndef::env-site,env-github[]
+supporting `fetchSize`, `maxRows`, and timeout parameters. Although examples of input
+parameterization, dynamic queries, and parameter usage are provided, you can consult
+the xref:db/db-connector-query.adoc[SELECT operation documentation]
+for more information.
+
+== Insert Example
+
+Suppose there is a database schema named *Products* that has a table named
+*electronic*. *electronic* contains columns named *id*, *name*, *description*, *price*, and *discount*.
+The following SQL statement creates the table:
+
+[source,sql,linenums]
+----
+CREATE TABLE electronic(
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(100),
+    description VARCHAR(255),
+    price SMALLINT,
+    discount TINYINT
+)
+----
+
+For an example on creating a table using the Database connector, check how to
+xref:db/db-connector-ddl.adoc[Execute DDL Statements].
+
+Given that the table is already created and that a there is a
+xref:db/db-configure-connection.adoc[connection configuration] named `dbConfig`,
+the following operation inserts a new record in the *electronic* table:
 
 [source,xml,linenums]
 ----
 <db:insert config-ref="dbConfig">
   <db:sql>
-    INSERT INTO PLANET(POSITION, NAME, DESCRIPTION) VALUES (777, 'Pluto', :description)
+    INSERT INTO electronic(name, description, price, discount)
+    VALUES ('Coffee Machine', 'Model: XYZ99. Uses small size capsules.', 120, 5)
   </db:sql>
-  <db:input-parameters>
-    #[
-    {'description' : payload}
-    ]
-  </db:input-parameters>
 </db:insert>
-view rawinsert.xml hosted with ❤ by GitHub
+----
+
+Note that the *id* parameter is not passed in the query because it is an
+autoincrement key that is generated automatically by the database.
+
+[NOTE]
+=============
+To benefit from the advantages of
+xref:db/db-connector-query.adoc#_use_input_parameters_to_protect_database_queries[input parameters],
+the equivalent query should be written as follows:
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig">
+    <db:sql>
+        INSERT INTO electronic(name, description, price, discount)
+        VALUES (:name, :description, :price, :discount)
+    </db:sql>
+    <db:input-parameters>#[{
+        name: 'Coffee Machine',
+        description: 'Model:XYZ99. Uses small size capsules.',
+        price: 120,
+        discount: 5}]
+    </db:input-parameters>
+</db:insert>
+----
+=============
+
+The `insert` operation will output a `Statement Result` object that contains 2 fields: a _affectedRows_
+integer that indicates how many rows were affected by the query, and a _generatedKeys_ map
+which contains the autogenerated keys. Given this example, the output is as follows:
+
+[source,json,linenums]
+----
+{
+  "affectedRows": 1,
+  "generatedKeys": {
+
+  }
+}
+----
+
+The _generatedKeys_ map is empty, although a row is successfully added to the table.
+This is because this information is disabled by default in order to avoid large overheads.
```

Comment:
Does "This" refer to the empty map or the row added? Clarify.

in order to > to

avoid large overheads > avoid overhead

Reply:
"The map is empty because"; accepted other feedback.